python/pyarrow/compute.py

```diff
@@ -33,7 +33,7 @@
     AssumeTimezoneOptions,
     CastOptions,
     CountOptions,
-    CumulativeSumOptions,
+    CumulativeOptions,
```

Comment:
@jorisvandenbossche Does this need an alias?
```suggestion
    CumulativeOptions,
    CumulativeOptions as CumulativeSumOptions,
```

Reply:
For backwards compatibility, that's probably useful, yes (also on the C++ side this was done). 

(although we should maybe actually add it as its own class and deprecate it, so we can remove the alias at some point)

Reply:
Follow up issue here https://github.com/apache/arrow/issues/36240